odoo/addons/base/migrations/13.0.1.3/post-migration.py

```diff
@@ -1,52 +1,58 @@
 # Copyright 2020 Andrii Skrypka
 # Copyright 2020 Opener B.V. (stefan@opener.amsterdam)
+# Copyright 2019-2020 Tecnativa - Pedro M. Baeza
+# Copyright 2020 ForgeFlow
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
-
-from psycopg2 import sql
+import pytz
+from datetime import datetime
 from openupgradelib import openupgrade
+from odoo import fields
 
-
-def fix_res_lang_url_code(env):
-    """ Url code is required. """
-    for lang in env['res.lang'].with_context(active_test=False).search([('url_code', '=', False)]):
-        lang.url_code = lang.iso_code or lang.code
+ACTIONS_TABLES = {
+    "ir_act_url",
+    "ir_act_window",
+    "ir_actions",
+    "ir_act_client",
+    "ir_act_report_xml",
+    "ir_act_server",
+}
 
 
-def fix_res_partner_image(env):
-    ResPartner = env['res.partner']
+def rename_res_partner_image(env):
+    res_partner = env['res.partner']
     attachments = env['ir.attachment'].search([
         ('res_model', '=', 'res.partner'),
         ('res_field', '=', 'image'),
         ('res_id', '!=', False),
     ])
     for attachment in attachments:
-        ResPartner.browse(attachment.res_id).image_1920 = attachment.datas
+        res_partner.browse(attachment.res_id).image_1920 = attachment.datas
 
 
-def restore_res_lang_week_start(env):
-    legacy_name = openupgrade.get_legacy_name('week_start')
-    openupgrade.logged_query(
+def res_lang_week_start_map_values(env):
+    openupgrade.map_values(
         env.cr,
-        """
-        SELECT id, %s
-        FROM res_lang
-        WHERE %s IS NOT NULL
-        """ % (legacy_name, legacy_name),
+        openupgrade.get_legacy_name("week_start"),
+        "week_start",
+        [(1, "1"), (2, "2"), (3, "3"), (4, "4"), (5, "5"), (6, "6"), (7, "7")],
+        table="res_lang",
     )
-    grouped_by_week_start = {}
-    for record in env.cr.fetchall():
-        grouped_by_week_start.setdefault(record[1], []).append(record[0])
-    for week_start, lang_ids in grouped_by_week_start.items():
-        env['res.lang'].browse(lang_ids).write({'week_start': str(week_start)})
-
 
-@openupgrade.migrate()
-def migrate(env, version):
-    fix_res_lang_url_code(env)
-    restore_res_lang_week_start(env)
-    openupgrade.load_data(env.cr, 'base', 'migrations/13.0.1.3/noupdate_changes.xml')
-    fix_res_partner_image(env)
 
+def ir_actions_binding_type_views(env):
+    # Convert obsolete 'action_form_only' binding type to binding_view_types
+    for table in ACTIONS_TABLES:
```

Comment:
The previous version only updated ir_actions on purpose because the other tables inherit from it on the SQL level. Did that not work as I expected?

Reply:
I only checked that the other actions have the same columns, so I thought that better update those columns by query instead of let odoo handle it.

Reply:
Indeed it's that way:

https://github.com/OCA/OpenUpgrade/blob/d607b670588e96917f89d8b6221bcc65e48345f3/odoo/addons/base/data/base_data.sql#L9-L13

so I'm reverting this change.

Reply:
@pedrobaeza do I understand correctly that this code will be reverted only to update ir_actions?

Reply:
Yes, pushing in a few minutes